memory/libhugetlbfs.py

```diff
@@ -41,19 +41,23 @@ def setUp(self):
         # Check for basic utilities
         sm = SoftwareManager()
         detected_distro = distro.detect()
-        if not sm.check_installed("gcc") and not sm.install("gcc"):
-            self.error('Gcc is needed for the test to be run')
-        if not sm.check_installed("make") and not sm.install("make"):
-            self.error('make is needed for the test to be run')
-        if detected_distro.name == "SuSE":
-            if not sm.check_installed("git-core") and not sm.install("git-"
-                                                                     "core"):
-                self.error('git is needed for the test to be run')
+        deps = ['gcc', 'make', 'git']
+
+        if detected_distro.name == "Ubuntu":
+            deps += ['libpthread-stubs0-dev']
+        elif detected_distro.name == "SuSE":
+            deps += ['glibc-devel-static']
         else:
-            if not sm.check_installed("git") and not sm.install("git"):
-                self.error('git is needed for the test to be run')
+            deps += ['glibc-static']
+
+        for package in deps:
+            if detected_distro.name == "SuSE" and package == 'git':
```

Comment:
It is better to do this check before, where you are doing a distro check for pthread and glibc.